### PR TITLE
Bug fix/audio fixes

### PR DESCRIPTION
**Pull request by Lord-Kamina**
### What does this PR do?

Chiefly, fixes the issue with Pa_OpenStream reporting invalid number of channels in windows. But it also fixes a couple other things I found along the way.

Succintly:

- Implement an additional member to the Devices to make sure we refer to them by their global index, instead of the one relative to their respective backend.
- Change the way the "mics" parameter is parsed from the schema.
- Correct a bug that would result in always using the first device when a schema or config entry did not have a "dev" parameter.
- Eliminate (I think) the audio artifacts when starting/restarting playback.

### Closes Issue(s)

#327 (at least, I think)

### More

- [ ] Added/updated documentation

### CODE REVIEW

File: game/CMakeLists.txt [hunk 1]
```diff
@@ -144,6 +144,7 @@ if(UNIX AND NOT APPLE)
 endif()
 
 if(WIN32)
+	add_definitions("-DEPOXY_SHARED")
```
Comment (Baklap4) [inline game/CMakeLists.txt]: Welcome back old friend <3

File: game/libda/portaudio.hpp [hunk 2]
```diff
@@ -118,6 +120,15 @@ namespace portaudio {
 			}
 			throw std::runtime_error("No such device.");
 		}
+		DeviceInfo const& findByChannels(bool output, unsigned num) {
+			// Try name search with full match
+			for (auto const& dev: devices) {
+			unsigned reqChannels = output ? dev.out : dev.in;
```
Comment (Baklap4) [inline game/libda/portaudio.hpp]: Indents

File: game/libda/portaudio.hpp [hunk 3]
```diff
@@ -223,6 +234,9 @@ namespace portaudio {
 		  PaStreamCallback* callback = NULL,
 		  void* userData = NULL)
 		{
+			if (output != nullptr) {
```
Comment (nieknooijens) [inline game/libda/portaudio.hpp]: little inconsistency here, you're using nullptr, but elsewhere NULL is used. might be a good idea to put these in the code standard.

Comment (Lord-Kamina) [reply game/libda/portaudio.hpp]: From what I understand, nullptr is generally to be preferred when working with pointers. So IMO we should update those other NULLs.

Comment (Lord-Kamina) [reply game/libda/portaudio.hpp]: P.S. I already did it but will open a new PR.

**Comment by nieknooijens** [comment]
indents have been fixed so @Baklap4 you can approve now.

**Pull request merged**